File: solutions/kotlin/difference-of-squares/1/src/main/kotlin/DifferenceOfSquares.kt
class Squares(val number: Int) {
    val sumOfSquares = (1..number).map{ it * it }.sum()
    val sum = (1..number).sum()
    val squareOfSum = sum * sum
    val difference = squareOfSum - sumOfSquares
    
    fun sumOfSquares(): Int {
        return sumOfSquares
    }

    fun squareOfSum(): Int {
        return squareOfSum
    }

    fun difference(): Int {
        return difference
    }
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) (r int) {
	if n <= 0 {
        return
    }
    r = n * n * (n+1) * (n+1) / 4
    return
}

func SumOfSquares(n int) (r int) {
	if n <= 0 {
        return
    }
    for i := 1; i <= n; i++ {
        r += i * i
    }
    return
}

func Difference(n int) (r int) {
	if n <= 0 {
        return
    }
    return SquareOfSum(n) - SumOfSquares(n)
}
